include/zephyr/drivers/display.h

```diff
@@ -193,6 +193,23 @@ typedef int (*display_set_orientation_api)(const struct device *dev,
 					   const enum display_orientation
 					   orientation);
 
+/**
+ * @typedef display_set_scroll_area_api
+ * @brief Callback API to set scrolling used by the display
+ * See display_set_scroll_area() for argument description
+ */
+typedef int (*display_set_scroll_area_api)(const struct device *dev,
+						uint16_t tfa, 
+						uint16_t bfa);
+
+/**
+ * @typedef display_set_scroll_api
+ * @brief Callback API to scroll used by the display
+ * See display_set_scroll() for argument description
+ */
+typedef int (*display_set_scroll_api)(const struct device *dev,
+						uint16_t val);
+
```

Comment:
Motivation for scrolling came from trying to eliminate screen tear. Using already built in functionality on the display driver allows processing power to be saved, screen tear is eliminated and scrolling to be done faster and smoother.

Tbh, as far as I've seen a lot of samples are missing from the display section. I was also planning on adding the lvgl demo apps to be used as samples. I could also make a sample using this in addition.

Reply:
Since I want to add the lvgl sample in also, I was thinking what the best fodler structure would be?
Something like this?:

display/lvgl/scroll             - I'd currently put it under lvgl since i was using lvgl. but i guess a non lvgl version would make more sense
display/scroll

and the rest of the lvgl demo applications could become:
the current basic sample would move under there as well

display/lvgl/basic
display/lvgl/benchmark
display/lvgl/keypad_encoder
display/lvgl/music
display/lvgl/stress
display/lvgl/widgets